### ISSUE TITLE
jpcw-register-explicit.cml does not typecheck

### ISSUE THREAD

**Issue by joey-coleman** [symphonytool/symphony#5]
> The following bug was [originally reported](https://sourceforge.net/p/compassresearch/bugs/5/) on Sourceforge by [anderskm](https://sourceforge.net/u/anderskm/profile/), 2013-03-07 11:13:52.717000:

It gives back the following error message in eclipse:

Multiple markers found on this line:
- Expected 1 args, 0 were found.  
- Could not determine type for INIT().  

The attached screenshot shows locations of this error 

It seems to have problems with operations that take no arguments : () => ()

**Comment by joey-coleman** [comment]
> [Comment](https://sourceforge.net/p/compassresearch/bugs/5/#50c0) by [anderskm](https://sourceforge.net/u/anderskm/profile/), 2013-03-07 11:15:00.168000:
- **assigned_to**: Rasmus Lauritsen

**Comment by joey-coleman** [comment]
> [Comment](https://sourceforge.net/p/compassresearch/bugs/5/#fd75) by [anderskm](https://sourceforge.net/u/anderskm/profile/), 2013-03-07 11:18:59.715000:
- **summary**: jpcw-register-explicit.cml does not typecheck as it should --> jpcw-register-explicit.cml does not typecheck
- Description has changed:

Diff:

```

--- old
+++ new
@@ -5,3 +5,5 @@
   - Could not determine type for INIT().  

 The attached screenshot shows locations of this error 
+
+It seems to have problems with operations that take no arguments : () => ()

```

**Comment by joey-coleman** [comment]
> [Comment](https://sourceforge.net/p/compassresearch/bugs/5/#6bdb) by [rasmuswl](https://sourceforge.net/u/rasmuswl/profile/), 2013-03-07 17:29:54.710000:

The bug has been fix in commit: [eb3f162446d6af92fd85dafe657572f4607ba8c1].

The parser was adding AVoidType to the argument list in the presence of no arguments which seemed quite natural, however for functions and operations the TC expects to list to be flat such that the parameters list is exactly the type arguments.

**Comment by joey-coleman** [comment]
> [Comment](https://sourceforge.net/p/compassresearch/bugs/5/#e1dc) by [rasmuswl](https://sourceforge.net/u/rasmuswl/profile/), 2013-03-07 17:30:47.428000:
- **status**: open --> closed

**Comment by joey-coleman** [comment]
> [Comment](https://sourceforge.net/p/compassresearch/bugs/5/#a751) by [rasmuswl](https://sourceforge.net/u/rasmuswl/profile/), 2013-03-07 17:30:47.638000:

Fixed on: [eb3f162446d6af92fd85dafe657572f4607ba8c1]
